mongo/options/findoptions.go

```diff
@@ -348,6 +352,8 @@ type FindOneOptions struct {
 
 	// If true, the cursor created by the operation will not timeout after a period of inactivity. The default value
```

Comment:
I believe `MaxAwaitTime` should also be deprecated. It is only applicable to tailable cursors. It is the maximum amount of time a tailable cursor will block on a getMore after the cursor is open.

Information about this is scattered, but the CRUD spect notes this in the FindOptions:

> The maximum amount of time for the server to wait on new documents to satisfy a tailable cursor
> query.
>
> This options only applies to aggregations which return a TAILABLE_AWAIT cursor. Drivers
> SHOULD always send this value, if the cursor is not a TAILABLE_AWAIT cursor the server will
> ignore it.
>
> note this option is an alias for `maxTimeMS`, used on `getMore` commands
> note this option is not set on the `aggregate` command

The [manual notes](https://docs.mongodb.com/manual/reference/glossary/#term-tailable-cursor) that tailable cursors are only currently applicable to capped collections.

Reply:
Yeah, since `MaxAwaitTime` is only applicable to when a cursor is created (albeit a tailable one), sounds like we can deprecate it. Done.